### PR TITLE
[melodic] pass ROS_BUILD_SHARED_LIBS to use visibility control properly.

### PR DESCRIPTION
**Pull request by seanyen**
Many ROS packages adopt this [visibility control header](http://wiki.ros.org/win_ros/Contributing/Dll%20Exports) to make the packages Windows-compatible. And the visibility control is conditionally guarded by `ROS_BUILD_SHARED_LIBS` which is defined by [catkin](https://github.com/ros/catkin/blob/kinetic-devel/cmake/tools/libraries.cmake#L18) in CMake file.

However, for `sip` module, it is running its own Makefile generated by `sip`, and this important definition (`ROS_BUILD_SHARED_LIBS`) is not automatically inherited. In such case, when it is building the `sip` module and ROS visibility control headers are in use there, it doesn't use the expected visibility control code path.

### CODE REVIEW

**Comment by seanyen** [comment]
@dirk-thomas This is ready for review and merge. Thanks!

**Comment by seanyen** [comment]
@dirk-thomas Friendly ping.

**Comment by seanyen** [comment]
@dirk-thomas Following up. Any feedback would be appreciated.

File: cmake/sip_helper.cmake [hunk 1]
```diff
@@ -79,14 +79,19 @@ function(build_sip_binding PROJECT_NAME SIP_FILE)
     set(LIBRARY_DIRS ${${PROJECT_NAME}_LIBRARY_DIRS})
     set(LDFLAGS_OTHER ${${PROJECT_NAME}_LDFLAGS_OTHER})
 
+    set(EXTRA_DEFINES "")
+    if(DEFINED BUILD_SHARED_LIBS)
```
Comment (dirk-thomas) [inline cmake/sip_helper.cmake:83]: `DEFINED` doesn't seem sufficient to check. This condition would be `TRUE` even if the CMake variable `BUILD_SHARED_LIBS` is `FALSE`.

Comment (seanyen) [reply cmake/sip_helper.cmake]: Thank for the catch. I corrected the check now.

Comment (dirk-thomas) [reply cmake/sip_helper.cmake]: Without the `DEFINED` check the code is accessing a potentially undefined variable (which results in a CMake warning if that warning is enabled). Please keep the old check with `DEFINED` as well as the new check for testing its actual value.

Comment (seanyen) [reply cmake/sip_helper.cmake]: Got it. I added both as the check.

**Comment by dirk-thomas** [comment]
Thanks for the patch.

**Pull request merged**